clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteringCacheManagerServiceConfigurationParser.java

```diff
@@ -0,0 +1,466 @@
+      parent.appendChild(connectionTimeoutElem);
+    }
+  }
+
+  private Element createTimeoutElement(Document doc, String timeoutName, Duration timeout) {
+    Element retElement = null;
+    if (READ_TIMEOUT_ELEMENT_NAME.equals(timeoutName)) {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + READ_TIMEOUT_ELEMENT_NAME);
+    } else if (WRITE_TIMEOUT_ELEMENT_NAME.equals(timeoutName)) {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + WRITE_TIMEOUT_ELEMENT_NAME);
+    } else {
+      retElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CONNECTION_TIMEOUT_ELEMENT_NAME);
+    }
+    retElement.setAttribute(UNIT_ATTRIBUTE_NAME, DEFAULT_UNIT_ATTRIBUTE_VALUE);
+    retElement.setTextContent(Long.toString(timeout.getSeconds()));
+    return retElement;
+  }
+
+  private Element processServerSideElements(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element serverSideConfigurationElem = createServerSideConfigurationElement(doc, clusteringServiceConfiguration);
+
+    if (clusteringServiceConfiguration.getServerConfiguration() != null) {
+      ServerSideConfiguration serverSideConfiguration = clusteringServiceConfiguration.getServerConfiguration();
+      String defaultServerResource = serverSideConfiguration.getDefaultServerResource();
+      if (!(defaultServerResource == null || defaultServerResource.trim().length() == 0)) {
+        Element defaultResourceElement = createDefaultServerResourceElement(doc, defaultServerResource);
+        serverSideConfigurationElem.appendChild(defaultResourceElement);
+      }
+      Map<String, ServerSideConfiguration.Pool> resourcePools = serverSideConfiguration.getResourcePools();
+      if (!(resourcePools == null || resourcePools.isEmpty())) {
+        resourcePools.forEach(
+          (key, value) -> {
+            Element poolElement = createSharedPoolElement(doc, key, value);
+            serverSideConfigurationElem.appendChild(poolElement);
+          }
+        );
+      }
+    }
+    return serverSideConfigurationElem;
+  }
+
+  private Element createServerSideConfigurationElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element serverSideConfigurationElem = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + SERVER_SIDE_CONFIG);
+    serverSideConfigurationElem.setAttribute(AUTO_CREATE_ATTRIBUTE_NAME, Boolean.toString(clusteringServiceConfiguration
+      .isAutoCreate()));
+    return serverSideConfigurationElem;
+  }
+
+
+  private Element createSharedPoolElement(Document doc, String poolName, ServerSideConfiguration.Pool pool) {
+    Element poolElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + SHARED_POOL_ELEMENT_NAME);
+    poolElement.setAttribute(NAME_ATTRIBUTE_NAME, poolName);
+    String from = pool.getServerResource();
+    if (!(from == null || from.trim().length() == 0)) {
```

Comment:
Why are we declaring strings of white space characters to be the same as requesting the default pool?

Reply:
Can a pool-resource be such that it is defined only using white spaces? If not, then pointing it to default one might be an option. Another way can deal with it is: If null then go for default; otherwise if the from property is a space only string, throw exception. What do you suggest?

Reply:
Via xml maybe not, but programaticaly it probably works... regardless if you see it the only correct behaviors are "do the right thing" or "fail", "do something entirely different" is not a valid option.